sklearn/exceptions.py

```diff
@@ -154,3 +155,12 @@ class UndefinedMetricWarning(UserWarning):
     .. versionchanged:: 0.18
        Moved from sklearn.base.
     """
+
+
+class SklearnDeprecationWarning(DeprecationWarning):
```

Comment:
True, but did we ever promise explicitly such a strong backward compatibility? Exception types make sense, but warning types seem outside the realm of backwards compatibility as far as I am concerned. My feeling is that a whats_new entry would be good enough if we decide to have SklearnDeprecationWarning inherit from UserWarning.

In my mind disabling warnings is more a hack than anything else and should not be encouraged. Maybe people are doing it (cleaner execution logs seems like a possible motivation), I am not sure.